pkg/testfixtures/storage/postgres.go

```diff
@@ -46,88 +44,36 @@ func (p *postgresTestContainer) GetDatabaseSchemaVersion() int64 {
+			WithOccurrence(2).
+			WithStartupTimeout(5*time.Second),
+		),
+		testcontainers.WithHostConfigModifier(func(hostConfig *container.HostConfig) {
+			hostConfig.Tmpfs = map[string]string{"/var/lib/postgresql/data": ""}
+		}),
+		testcontainerspostgres.WithDatabase("defaultdb"),
+		testcontainerspostgres.WithUsername("postgres"),
+		testcontainerspostgres.WithPassword("secret"),
 	)
 	require.NoError(t, err)
-	t.Cleanup(func() {
-		dockerClient.Close()
-	})
+	t.Cleanup(func() { require.NoError(t, postgresContainer.Terminate(ctx)) })
 
-	allImages, err := dockerClient.ImageList(context.Background(), types.ImageListOptions{
-		All: true,
-	})
+	postgresHost, err := postgresContainer.Host(ctx)
 	require.NoError(t, err)
-
-	foundPostgresImage := false
-	for _, image := range allImages {
-		for _, tag := range image.RepoTags {
-			if strings.Contains(tag, postgresImage) {
-				foundPostgresImage = true
-				break
-			}
-		}
-	}
-
-	if !foundPostgresImage {
-		t.Logf("Pulling image %s", postgresImage)
-		reader, err := dockerClient.ImagePull(context.Background(), postgresImage, types.ImagePullOptions{})
-		require.NoError(t, err)
-
-		_, err = io.Copy(io.Discard, reader) // consume the image pull output to make sure it's done
-		require.NoError(t, err)
-	}
-
-	containerCfg := container.Config{
-		Env: []string{
-			"POSTGRES_DB=defaultdb",
-			"POSTGRES_PASSWORD=secret",
-		},
-		ExposedPorts: nat.PortSet{
-			nat.Port("5432/tcp"): {},
-		},
-		Image: postgresImage,
-	}
-
-	hostCfg := container.HostConfig{
-		AutoRemove:      true,
-		PublishAllPorts: true,
-		Tmpfs:           map[string]string{"/var/lib/postgresql/data": ""},
-	}
-
-	name := fmt.Sprintf("postgres-%s", ulid.Make().String())
-
-	cont, err := dockerClient.ContainerCreate(context.Background(), &containerCfg, &hostCfg, nil, nil, name)
-	require.NoError(t, err, "failed to create postgres docker container")
-
-	t.Cleanup(func() {
-		t.Logf("stopping container %s", name)
-		timeoutSec := 5
-
-		err := dockerClient.ContainerStop(context.Background(), cont.ID, container.StopOptions{Timeout: &timeoutSec})
-		if err != nil && !client.IsErrNotFound(err) {
-			t.Logf("failed to stop postgres container: %v", err)
-		}
-
-		t.Logf("stopped container %s", name)
-	})
-
-	err = dockerClient.ContainerStart(context.Background(), cont.ID, container.StartOptions{})
-	if err != nil {
-		t.Fatalf("failed to start postgres container: %v", err)
-	}
-
-	containerJSON, err := dockerClient.ContainerInspect(context.Background(), cont.ID)
+	postgresPort, err := postgresContainer.MappedPort(ctx, "5432/tcp")
 	require.NoError(t, err)
 
-	m, ok := containerJSON.NetworkSettings.Ports["5432/tcp"]
-	if !ok || len(m) == 0 {
-		t.Fatalf("failed to get host port mapping from postgres container")
-	}
-
 	pgTestContainer := &postgresTestContainer{
-		addr:     fmt.Sprintf("localhost:%s", m[0].HostPort),
+		addr:     net.JoinHostPort(postgresHost, postgresPort.Port()),
```

Comment:
Same about the `ConnectionString` method of the postgres container

Reply:
it will need refactoring a bit, so I will consider in next PRs